modules/mono/csharp_script.cpp

```diff
@@ -113,6 +113,9 @@ void CSharpLanguage::init() {
 	BindingsGenerator::handle_cmdline_args(cmdline_args);
 #endif
 
+	GLOBAL_DEF("dotnet/project/assembly_name", "");
+	GLOBAL_DEF("dotnet/project/solution_directory", "");
```

Comment:
I moved it to the `CSharpLanguage` constructor but the `GDMono` constructor also seems like a good place to me. GDScript seems to register some properties in the `GDScriptLanguage` constructor which is what made me go with `CSharpLanguage`.

Reply:
This is the right place. The solution dir property could be surrounded with `TOOLS_ENABLED`.